src/main/java/teammates/ui/controller/StudentCourseJoinAction.java

```diff
@@ -34,8 +35,12 @@ public ActionResult execute() throws EntityDoesNotExistException {
         
         String confirmUrl = Const.ActionURIs.STUDENT_COURSE_JOIN_AUTHENTICATED 
                 + "?" + Const.ParamsNames.REGKEY + "=" + regkey 
-                + "&" + Const.ParamsNames.NEXT_URL + "=" + nextUrl;
-        data = new StudentCourseJoinConfirmationPageData(account, student, confirmUrl, Logic.getLogoutUrl(confirmUrl));
+                + "&" + Const.ParamsNames.NEXT_URL + "=" + Sanitizer.sanitizeForNextUrl(nextUrl);
+        String studentNewCourse = getRequestParamValue(Const.ParamsNames.STUDENT_NEW_COURSE);
+        boolean isUserLoggedInAsAnotherStudent = studentNewCourse != null;
+        data = new StudentCourseJoinConfirmationPageData(account, student, confirmUrl,
+                                                         Logic.getLogoutUrl(Sanitizer.sanitizeForNextUrl(confirmUrl)),
+                                                         isUserLoggedInAsAnotherStudent, studentNewCourse);
```

Comment:
It works, and looks like it is the better alternative after all. Should be good to go now, I guess?